File: pglite/src/pglite-custom/release/core/runtimeState.ts
import type {
	DataManifest,
	RuntimeModule,
	RuntimeState,
} from "./runtimeTypes.js";

const EMPTY_MANIFEST: DataManifest = {
	files: [],
	remotePackageSize: 0,
};

export function createRuntimeState(
	moduleArg: Partial<RuntimeModule>,
	manifest: DataManifest = EMPTY_MANIFEST,
): RuntimeState {
	return {
		moduleArg,
		module: null,
		asmHandlers: new Map(),
		emJsHandlers: new Map(),
		dataManifest: manifest,
		diagnostics: [],
	};
}

export function bindRuntimeModule(
	state: RuntimeState,
	module: RuntimeModule,
): RuntimeModule {
	state.module = module;
	return module;
}

File: src/main/java/io/github/hidekatsu_izuno/pglite_jdbc/pglite/release/core/runtimeState.java
package io.github.hidekatsu_izuno.pglite_jdbc.pglite.release.core;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class runtimeState {
    private runtimeState() {}

    public static runtimeTypes.RuntimeState createRuntimeState(
        Map<String, Object> moduleArg,
        runtimeTypes.DataManifest manifest
    ) {
        return new runtimeTypes.RuntimeState(
            moduleArg != null ? moduleArg : new HashMap<>(),
            null,
            new HashMap<>(),
            new HashMap<>(),
            manifest != null ? manifest : runtimeTypes.DataManifest.empty(),
            new ArrayList<>()
        );
    }

    public static runtimeTypes.RuntimeModule bindRuntimeModule(
        runtimeTypes.RuntimeState state,
        runtimeTypes.RuntimeModule module
    ) {
        state.setModule(module);
        return module;
    }
}
